modules/graphql_core/src/Plugin/GraphQL/Fields/EntityQuery/EntityQuery.php

```diff
@@ -409,8 +419,22 @@ protected function isNullOperator($operator) {
    *   TRUE if the given operator is a range operator, FALSE otherwise.
    */
   protected function isRangeOperator($operator) {
-    $null = ["BETWEEN", "NOT BETWEEN"];
-    return in_array($operator, $null);
+    $range = ["BETWEEN", "NOT BETWEEN"];
+    return in_array($operator, $range);
+  }
+
+  /**
+   * Checks if an operator is potentially unsafe.
+   *
+   * @param string $operator
+   *   The query operator to check against.
+   *
+   * @return bool
+   *   TRUE if the given operator is an unsafe operator, FALSE otherwise.
+   */
+  protected function isUnsafeOperator($operator) {
+    $unsafe = ["<", "<=", ">", ">=", "LIKE", "NOT LIKE", "BETWEEN", "NOT BETWEEN"];
```

Comment:
instead of defining a list of unsafe operators you should define a list of safe operators. That way you limit it to a clearly allowed list and nothing else.